app/src/main/java/edu/grinnell/kdic/MainActivity.java

```diff
@@ -261,14 +270,11 @@ private void setupNavigation() {
         mNavigationToolbar.setNavigationOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                mDrawerLayout.openDrawer(GravityCompat.START);
+                mDrawerLayout.openDrawer(START);
             }
         });
 
-        // set up mBackStack
         mBackStack = new Stack<>();
```

Comment:
Aha! So you're initializing `mBackStack` here, which is good. But you never free it after use for recycling. You should nullify it in `onDestroy()` else you're likely to leak some memory